simba/simulate.py

```diff
@@ -22,13 +23,23 @@ def simulate(args):
     :return: final schedule and scenario
     :rtype: tuple
     """
-    # The DataContainer stores various input data.
-    data_container = DataContainer().fill_with_args(args)
-
-    schedule, args = pre_simulation(args, data_container)
-    scenario = schedule.run(args)
-    schedule, scenario = modes_simulation(schedule, scenario, args)
-    return schedule, scenario
+    if vars(args).get("load_pickle"):
+        # load pickle file: skip pre_simulation
+        if isinstance(args.mode, list):
+            first_mode = args.mode[0]
+        else:
+            first_mode = args.mode
+        assert first_mode == "load_pickle", "Load pickle: first mode must be load_pickle"
+        # schedule and scenario read out from pickle file in first mode
+        # DataContainer is part of schedule
+        schedule = None
+        scenario = "pickle"  # must not be None
```

Comment:
with the above change i would like
 `
assert args.mode[0] == "load_pickle"
schedule, scenario = load_pickle(None, None, None)
args.mode = args.mode[1:]
`
i really dislike changing type of scenario, even for just a short while.